src/nprintml/automl.py

```diff
@@ -0,0 +1,143 @@
+import pathlib
+import matplotlib.pyplot as plt
+
+import numpy as np
+import pandas as pd
+import autogluon as ag
+from autogluon import TabularPrediction as task
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.metrics import roc_curve, auc, precision_recall_curve, \
+        average_precision_score, confusion_matrix, ConfusionMatrixDisplay
+
+import seaborn as sns
+sns.set_style("ticks")
+sns.set_context(rc={"lines.linewidth": 1.75})
+
+
+class AutoML:
+    def __init__(self, ml_data, outpath):
+        self.data = ml_data
+        self.outpath = outpath
+        self.qd = {
+          0: 'medium_quality_faster_train',
+          1: 'good_quality_faster_inference_only_refit',
+          2: 'high_quality_fast_inference_only_refit',
+          3: 'best_quality',
+          }
+
+    def run(self, test_size=.30, eval_metric='accuracy', quality=0,
+            time_limits=5*60, n_threads=1):
```

Comment:
We're going to need cli flags for

- test size: 0.00 - 1.00 (size of test split)
- eval_metric: metric to optimize for, [available list:](https://auto.gluon.ai/stable/api/autogluon.task.html#tabular)
- quality: [available list:](https://auto.gluon.ai/stable/api/autogluon.task.html#tabular)
- time_limits: max time limit in seconds for each model
- n_threads: number of threads for training